src/utils/GraphhopperUtils.js

```diff
@@ -148,15 +212,31 @@ async function fetchRoutes(end: string, start: string, departureTimeDateNow: str
       true,
     ),
   ]).then((vals) => {
-    busRouteRequest = vals[0];
-    walkingRouteRequest = vals[1];
+    busRouteNowRequest = vals[0];
+    busRouteBufferedRequest = vals[1];
+    walkingRouteRequest = vals[2];
   });
 
-  if (busRouteRequest && busRouteRequest.statusCode < 300) {
-    busRoutes = JSON.parse(busRouteRequest.body);
+  if (busRouteNowRequest && busRouteNowRequest.statusCode < 300) {
+    busRoutes = JSON.parse(busRouteNowRequest.body).paths;
+  } else {
+    LogUtils.log(
+      busRouteNowRequest && busRouteNowRequest.body,
+      getGraphhopperBusParams(end, start, departureTimeDateNow, isArriveByQuery),
+      `Routing failed: ${GHOPPER_BUS || 'undefined graphhopper bus env'}`,
+    );
+  }
+
+  if (busRouteBufferedRequest && busRouteBufferedRequest.statusCode < 300) {
+    const bufferedBusRoutes = JSON.parse(busRouteBufferedRequest.body).paths;
+    const validBusRoutes = bufferedBusRoutes.filter((bufferedRoute) => {
+      const isDuplicateRoute = busRoutes.find(busRoute => busRoutesAreEqual(bufferedRoute, busRoute)) !== undefined;
```

Comment:
probably cleaner to do:

`busRoutes.find(busRoute => busRoutesAreEqual(bufferedRoute, busRoute)) || false`;

Reply:
wouldn't that find the `busRoutes` that are duplicates?